metals/src/main/scala/scala/meta/metals/MetalsServices.scala

```diff
@@ -441,6 +442,22 @@ class MetalsServices(
   }
   private def sbtExec(): Unit = sbtExec(latestConfig().sbt.command)
 
+  private val loadPluginJars: Coeval[List[AbsolutePath]] = Coeval {
+    Jars.fetch("ch.epfl.scala", "load-plugin_2.12", "0.1.0+2-496ac670")
+  }
+
+  private def sbtExecWithMetalsPlugin(commands: String*): Unit = {
+    val metalsPluginModule =
+      ModuleID("org.scalameta", "sbt-metals", "0.1.0-M1+84-a3ffb91c")
+    val metalsPluginRef = "scala.meta.sbt.MetalsPlugin"
+    val loadPluginClasspath = loadPluginJars.value.mkString(":")
+    val loadCommands = Seq(
+      s"apply -cp ${loadPluginClasspath} ch.epfl.scala.loadplugin.LoadPlugin",
+      s"""if-absent ${metalsPluginRef} "load-plugin ${metalsPluginModule} ${metalsPluginRef}"""",
```

Comment:
@Duhemm it would be nice to check the exact version of the loaded plugin (and fail if it doesn't match. Would it be possible to do in a similar way `if-absent` works but with something like this to get all loaded plugin module IDs:

```scala
        val pluginsModules = for {
          (_, unit) <- structure.units
          report <- unit.unit.plugins.pluginData.report.toSeq
          module <- report.allModules
        } yield module
```

(modifying `pluginClasses`) 

I'll try tomorrow, but does it make sense in general.

Reply:
OK, I tried it and my first guess is wrong. The code above lists some modules, but it's not related to the plugins. I'm not sure what it is, but I think just the sbt dependencies itself.

I'd like to use it to detect if user has a different version of the plugin loaded (as if they add it globally) and as we can't get any useful result from running a command (on sbt server), I would make it just fail, get this failure and emit a warning message. It could take some other commands to run if the check passes, similarly to `if-absent`, I'm not sure if it's useful.